src/operator/src/statement.rs

```diff
@@ -390,8 +424,8 @@ impl StatementExecutor {
 
     #[tracing::instrument(skip_all)]
     async fn plan_exec(&self, stmt: QueryStatement, query_ctx: QueryContextRef) -> Result<Output> {
-        let plan = self.plan(&stmt, query_ctx.clone()).await?;
-        self.exec_plan(plan, query_ctx).await
+        let timeout = derive_timeout(&stmt, &query_ctx);
+        self.plan_exec_inner(stmt, query_ctx, timeout).await
```

Comment:
We can put the timeout logic in this method. The `plan_exec_inner()` only plans and executes the statement.
```suggestion
        let timeout = derive_timeout(&stmt, &query_ctx);
        match timeout {
            Some(timeout) => {
                let start = Instant::now();
                let output = tokio::time::timeout(timeout, self.plan_exec_inner()).await?;
                // compute remaining timeout......
                attach_timeout_stream(result?, remaining_timeout)
            },
            None => self.plan_exec_inner().await
        }
```